registry/remote/example_test.go

```diff
@@ -0,0 +1,371 @@
+}
+
+// ExampleRepository_Resolve_byDigest gives example snippets for resolving a digest.
+func ExampleRepository_Resolve_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // Resolve digest to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(descriptor.MediaType)
+	fmt.Println(descriptor.Digest)
+	fmt.Println(descriptor.Size)
+
+	// Output:
+	// application/vnd.oci.image.layer.v1.tar
+	// sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec
+	// 20
+}
+
+// ExampleRepository_Fetch_byTag gives example snippets for downloading a blob by tag.
+func ExampleRepository_Fetch_byTag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	tag := "latest"
+	descriptor, err := repo.Resolve(ctx, tag) // First resolve the tag to the descriptor
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// **You still need to validate the digest**
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example blob content
+}
+
+// ExampleRepository_Fetch_byDigest gives example snippets for downloading a blob by digest.
+func ExampleRepository_Fetch_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // We still need to resolve first, don't create a new descriptor with the digest, blob size is unknown
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+
+	// You can 1) fetch a whole blob
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// Fetch() doesn't provide validation, you need to do it **
```

Comment:
Try using terms defined in [BCP 14](https://tools.ietf.org/search/bcp14). In this case, callers **SHOULD** verify what they have fetched.

Reply:
This is really helpful